staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope.go

```diff
@@ -117,22 +170,30 @@ func (t *envelopeTransformer) TransformFromStorage(data []byte, context value.Co
 
 // TransformToStorage encrypts data to be written to disk using envelope encryption.
 func (t *envelopeTransformer) TransformToStorage(data []byte, context value.Context) ([]byte, error) {
-	newKey, err := generateKey(32)
-	if err != nil {
-		return nil, err
-	}
+	var (
+		err         error
+		transformer value.Transformer
+	)
 
-	encKey, err := t.envelopeService.Encrypt(newKey)
-	if err != nil {
-		// Do NOT wrap this err using fmt.Errorf() or similar functions
-		// because this gRPC status error has useful error code when
-		// record the metric.
-		return nil, err
+	// TODO(immutableT) Proactively generate a new DEK when the expiration period is
```

Comment:
I don't think this is necessary.
"Expiration" here is not a deadline after which DEK is unusable. It's simply an arbitrary caching window. It's OK to use it shortly after "expiry".

Reply:
I think "expiration" is very ambiguous.

Maybe give names to the phases somewhere and then it will become clear.
* phases being: (used for encrypted), (available for reading), (destroyed, data unreadable)
* active, passive, gone?
* working, retired, posthumous?
* shiny, tarnished, destroyed?

(the point of naming the last phase is to help people understand that it's not what we're talking about)